test/language/expressions/assignment/destructuring/obj-prop-nested-array-yield-expr.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    When a `yield` token appears within the Initializer of a nested
+    destructuring assignment and within a generator function body, it should
+    behave as a YieldExpression.
+es6id: 12.14.5.4
```

Comment:
Aww, I *just* finished typing up my rationale (`<sub>` tags and everything!). Posting it here for posterity I guess

```js
({ x: [x = yield] } = { x: [] });
```

- AssignmentPattern<sub>[Yield]</sub>
  - ArrayAssignmentPattern<sub>[?Yield]</sub>
    - **`[`** AssignmentElementList<sub>[?Yield]</sub> **`]`**
      - AssignmentElisionElement<sub>[?Yield]</sub>
        - Elision<sub>opt</sub> AssignmentElement<sub>[?Yield]</sub>
          - DestructuringAssignmentTarget<sub>[?Yield]</sub> Initializer<sub>[In,?Yield]opt</sub>

- Initializer<sub>[In, Yield]</sub>
  - **`=`** AssignmentExpression<sub>[?In, ?Yield]</sub>

- AssignmentExpression<sub>[In, Yield]</sub>
  - [+Yield] YieldExpression<sub>[?In]</sub>

- YieldExpression<sub>[In]</sub>
  - **`yield`**